torch/_torch_docs.py

```diff
@@ -3091,7 +3091,23 @@ def merge_dicts(*dicts):
            r"""
 flatten(input, start_dim=0, end_dim=-1) -> Tensor
 
-Flattens a contiguous range of dims in a tensor.
+Flattens :attr:`input`, or if :attr:`start_dim` or :attr:`end_dim` are specified, a range of :attr:`input`
```

Comment:
What if we combine the first two paragraphs like this:

"Flattens input by replacing its dimensions with a single dimension or, if start_dim and end_dim are specified, replacing the dimensions of input starting at start_dim and ending with end_dim with a single dimension. The order of elements in the tensor is unchanged."

Reply:
Its interesting to approach this from the dimension perspective, but we lose out on the meaning from "reshape/rearrange" here. Its no longer obvious that the number of elements is the same. We could do instead:

"Flattens :attr:`input` by reshaping it into a one-dimensional tensor with the order of elements unchanged. If :attr:`start_dim` or :attr:`end_dim` are passed, only dimensions starting with :attr:`start_dim` and ending  with :attr:`end_dim` are flattened.

Reply:
That sounds good but what about taking "tensor with the order of elements unchanged" and keeping that as a last sentence still like in the prior version. So then it'd be:

Flattens :attr:input by reshaping it into a one-dimensional tensor. If :attr:start_dim or :attr:end_dim are passed, only dimensions starting with :attr:start_dim and ending with :attr:end_dim are flattened. The order of elements in :attr:input is unchanged.